iroh-metrics/src/core.rs

```diff
@@ -0,0 +1,114 @@
+use prometheus_client::{encoding::text::encode, registry::Registry};
+use tokio::sync::RwLock;
+
+#[cfg(feature = "bitswap")]
+use crate::bitswap;
+#[cfg(feature = "gateway")]
+use crate::gateway;
+#[cfg(feature = "p2p")]
+use crate::p2p;
+#[cfg(feature = "resolver")]
+use crate::resolver;
+#[cfg(feature = "store")]
+use crate::store;
+
+lazy_static! {
+    pub(crate) static ref CORE: RwLock<Core> = RwLock::new(Core::default());
+}
+
+pub(crate) struct Core {
+    enabled: bool,
```

Comment:
I was unsure about the best way to approach the enabled/disabled thing especially about doing it at compile time.

Reply:
this seems like an interesting approach: https://docs.rs/libp2p/latest/libp2p/swarm/behaviour/toggle/struct.Toggle.html